rpm/0002-Add-D-Bus-interface-for-querying-client-process-deta.patch

```diff
@@ -0,0 +1,196 @@
+the client behind the proxy connection.
+
+Implement org.sailfishos.sailjailed.Identify() method call that returns
+pid, uid, and other details about the connected client.
+
+D-Bus policy configuration allowing/denying such method calls to be made
+is assumed to be defined elsewhere.
+---
+ flatpak-proxy.c | 143 +++++++++++++++++++++++++++++++++++++++++++++++-
+ 1 file changed, 142 insertions(+), 1 deletion(-)
+
+diff --git a/flatpak-proxy.c b/flatpak-proxy.c
+index 8bf9a4f..4282900 100644
+--- a/flatpak-proxy.c
++++ b/flatpak-proxy.c
+@@ -20,6 +20,7 @@
+ 
+ #include "config.h"
+ 
++#include <stdio.h>
+ #include <unistd.h>
+ #include <string.h>
+ 
+@@ -2306,6 +2307,142 @@ handle_deny:
+     queue_initial_name_ops (client);
+ }
+ 
++static gchar *
++get_pid_exe(gint32 pid)
++{
++  gchar *path = g_strdup_printf ("/proc/%d/exe", pid);
++  gchar *data = g_file_read_link (path, NULL);
++  g_free (path);
++  return data;
++}
++
++static gchar *
++get_pid_comm(gint32 pid)
++{
++  gchar *path = g_strdup_printf ("/proc/%d/comm", pid);
++  GFile *file = g_file_new_for_path (path);
++  gchar *data = NULL;
++  if (g_file_load_contents (file, NULL, &data, NULL, NULL, NULL))
++    data[strcspn (data, "\r\n")] = 0;
++  g_free (path);
++  g_object_unref (file);
++  return data;
++}
++
++static gchar *
++get_pid_cmdline(gint32 pid)
++{
++  gchar *path = g_strdup_printf ("/proc/%d/cmdline", pid);
++  GFile *file = g_file_new_for_path (path);
++  gchar *data = NULL;
++  if (g_file_load_contents (file, NULL, &data, NULL, NULL, NULL))
++    data[strcspn (data, "\n")] = 0;
++  g_free (path);
++  g_object_unref (file);
++  return data;
++}
++
++static gchar *
++get_pid_maps(gint32 pid)
++{
++  gchar *path = g_strdup_printf ("/proc/%d/maps", pid);
++  gchar *data = NULL;
++  FILE *file = fopen (path, "r");
++  if (file)
++    {
++      size_t size = 0;
++      char *text = NULL;
++      while (getline (&text, &size, file) >= 0)
++        {
++          if (!strstr (text, " r-xp "))
++            continue;
++          const char *mapped = strchr (text, '/');
++          if (!mapped)
++            continue;
++          if (!strstr (mapped, "/bin/"))
++            continue;
++          data = g_strndup (mapped, strcspn (mapped, "\n"));
++          break;
++        }
++      free (text);
++      fclose (file);
++    }
++  g_free (path);
++  return data;
++}
++
++static gboolean
++handle_proxy_method (FlatpakProxyClient *client, Header *header, Buffer *buffer)
++{
++  if (header->type != G_DBUS_MESSAGE_TYPE_METHOD_CALL)
++    return FALSE;
++  if (g_strcmp0 (header->interface, "org.sailfishos.sailjailed"))
++    return FALSE;
++  GDBusMessage *message = g_dbus_message_new_from_blob (buffer->data, buffer->size,
++                                                        G_DBUS_CAPABILITY_FLAGS_NONE, NULL);
++  if (message)
++    {
++      GDBusMessage *reply = NULL;
++      if (!g_strcmp0(header->member, "Identify"))
```

Comment:
nit: No space between method name and `(`.

Reply:
Fixed